src/dependency-manager/src/service-config/builder.ts

```diff
@@ -13,31 +15,68 @@ class MissingConfigFileError extends Error {
 }
 
 export class ServiceConfigBuilder {
-  static CONFIG_FILENAME = 'architect.json';
+  static getConfigPaths(input: string) {
+    return [
+      input,
+      path.join(input, 'architect.json'),
+      path.join(input, 'architect.yml'),
+      path.join(input, 'architect.yaml'),
+    ];
+  }
+
+  static buildFromPath(input: string): ServiceConfig {
+    const try_files = ServiceConfigBuilder.getConfigPaths(input);
```

Comment:
Looping over the files seems a little weird. You could check if the input is a file.
```
var fs = require('fs');
var stats = fs.statSync("c:\\dog.jpg");
console.log('is file ? ' + stats.isFile());
```

Reply:
The first item in the returned array is exactly the input. If it's a file that exists, we never reach the second array item.

Reply:
Why return a list though? Why not return the path given an input?

Reply:
At some point I need to check if the folder contains an architect.json, architect.yml, or architect.yaml unless I'm given a file path explicitly. The iterator handles all cases and prioritizes them as-is.